test/sql/test_temporal.py

```diff
@@ -0,0 +1,221 @@
+"""Tests for temporal table structure, including system and
+application versioning"""
+from sqlalchemy import Column
+from sqlalchemy import ForeignKeyConstraint
+from sqlalchemy import Integer
+from sqlalchemy import MetaData
+from sqlalchemy import Period
+from sqlalchemy import PrimaryKeyConstraint
+from sqlalchemy import schema
+from sqlalchemy import SystemTimePeriod
+from sqlalchemy import Table
+from sqlalchemy.sql.sqltypes import DATE
+from sqlalchemy.sql.sqltypes import TIMESTAMP
+from sqlalchemy.sql.sqltypes import VARCHAR
+from sqlalchemy.testing import AssertsCompiledSQL
+from sqlalchemy.testing import eq_
+from sqlalchemy.testing import fixtures
+from sqlalchemy.testing import is_
+
+
+class PeriodTest(fixtures.TestBase, AssertsCompiledSQL):
+    """Test the basic period construct"""
+
+    __dialect__ = "default"
+
+    def test_period(self):
+        m = MetaData()
+        t = Table(
+            "t",
+            m,
+            Column("start", TIMESTAMP),
+            Column("end", TIMESTAMP),
+            Period("test_period", "start", "end"),
```

Comment:
these would all be ApplicationPeriod or SystemPeriod, those are the two concrete classes

Reply:
Kind of relates to https://github.com/sqlalchemy/sqlalchemy/pull/7933#discussion_r852338879